pkg/util/etcdutil/migrate.go

```diff
@@ -41,34 +41,36 @@ func Migrate() error {
 	}
 	glog.Infof("boot-etcd pod IP is: %s", ip)
 
-	if err := createMigratedEtcdCluster(httpcli, apiserverAddr, ip); err != nil {
+	if err := createMigratedEtcdCluster(restClient, apiserverAddr, ip); err != nil {
 		glog.Errorf("fail to create migrated etcd cluster: %v", err)
 		return err
 	}
 
 	return checkEtcdClusterUp()
 }
 
-func listETCDCluster(host, ns string, httpClient *http.Client) (*http.Response, error) {
-	return httpClient.Get(fmt.Sprintf("%s/apis/coreos.com/v1/namespaces/%s/etcdclusters",
-		host, ns))
+func listETCDCluster(ns string, restClient restclient.Interface) restclient.Result {
+	uri := fmt.Sprintf("/apis/coreos.com/v1/namespaces/%s/etcdclusters", ns)
+	return restClient.Get().RequestURI(uri).Do()
 }
 
-func waitEtcdTPRReady(httpClient *http.Client, interval, timeout time.Duration, host, ns string) error {
+func waitEtcdTPRReady(restClient restclient.Interface, interval, timeout time.Duration, ns string) error {
 	err := wait.Poll(interval, timeout, func() (bool, error) {
-		resp, err := listETCDCluster(host, ns, httpClient)
-		if err != nil {
-			return false, err
+		res := listETCDCluster(ns, restClient)
+		if res.Error() != nil {
+			return false, res.Error()
 		}
-		defer resp.Body.Close()
 
-		switch resp.StatusCode {
+		var status int
+		res.StatusCode(&status) // this func sig is very odd...
```

Comment:
nit: not sure the comment here is super helpful - unless we need to track that it is odd :) ...